bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/LedgerEntry.java

```diff
@@ -56,25 +55,31 @@
     long getLength();
 
     /**
-     * Returns the content of the entry. This method can be called only once. While using v2 wire protocol this method
-     * will automatically release the internal ByteBuf.
+     * Returns the content of the entry.
      *
      * @return the content of the entry
-     * @throws IllegalStateException if this method is called twice
      */
     byte[] getEntry();
 
     /**
```

Comment:
What happens to the ownership of the returned ByteBuf? Is the caller responsible for releasing it? @sijie @zhaijack

Reply:
@ivankelly - This call doesn't retain any reference of the underneath bytebuf. if a caller releases the entry by #close(), it will release the underneath bytebuf. If the caller want to use the bytebuf beyond the lifecycle of this entry, it has to retain the buf by itself and then be responsible for releasing its retained refcnt. I have improved the comment in my subsequent change at #755 . 

https://github.com/apache/bookkeeper/pull/755/files

Reply:
ah, excellent, will check it out there.